theano/tensor/tests/test_extra_ops.py

```diff
@@ -33,6 +35,123 @@ def test_cpu_contiguous():
+                            searchsorted(self.x, self.v, sorter=sorter))
+        #  assert np.allclose(
+        #          np.searchsorted(self.a, self.b, sorter=None),
+        #          f(self.a, self.b, sorter=None))
+        self.assertRaises(ValueError, f,
+                          self.a[self.idx_sorted], self.b, None)
+
+    def test_searchsortedOp_on_float_sorter(self):
+        sorter = T.vector('sorter', dtype="float32")
+        self.assertRaises(TypeError, searchsorted,
+                          self.x, self.v, sorter=sorter)
+
+    def test_searchsortedOp_on_int_sorter(self):
+        compatible_types = ('int8', 'int16', 'int32', 'int64',)
+                            #  'uint8', 'uint16', 'uint32', 'uint64')
+        for dtype in compatible_types:
+            sorter = T.vector('sorter', dtype=dtype)
+            f = theano.function([self.x, self.v, sorter],
+                                searchsorted(self.x, self.v, sorter=sorter),
+                                mode="DebugMode", allow_input_downcast=True)
+            assert np.allclose(
+                    np.searchsorted(self.a, self.b, sorter=self.idx_sorted),
+                    f(self.a, self.b, self.idx_sorted))
+
+    def test_searchsortedOp_on_right_side(self):
+        f = theano.function([self.x, self.v],
+                            searchsorted(self.x, self.v, side='right'),
+                            mode="DebugMode")
+        assert np.allclose(
+                np.searchsorted(self.a, self.b, side='right'),
+                f(self.a, self.b))
+
+    def test_use_c_code(self):
+        f = theano.function([self.x, self.v], searchsorted(self.x, self.v),
+                            mode="FAST_RUN")
+        assert np.allclose(
+                np.searchsorted(self.a[self.idx_sorted], self.b),
+                f(self.a[self.idx_sorted], self.b))
+
+        f = theano.function([self.x, self.v], searchsorted(self.x, self.v),
+                            mode=theano.compile.Mode(linker="c",
+                            optimizer='fast_run'))
+        assert np.allclose(
+                np.searchsorted(self.a[self.idx_sorted], self.b),
+                f(self.a[self.idx_sorted], self.b))
+
+    def test_infer_shape(self):
+        # Test using default parameters' value
+        self._compile_and_check([self.x, self.v],
+                                [searchsorted(self.x, self.v)],
+                                [self.a[self.idx_sorted], self.b],
+                                self.op_class)
+
+        # Test parameter ``sorter``
+        sorter = T.vector('sorter', dtype="int64")
+        self._compile_and_check([self.x, self.v, sorter],
+                                [searchsorted(self.x, self.v, sorter=sorter)],
+                                [self.a, self.b, self.idx_sorted],
+                                self.op_class)
+
+        # Test parameter ``side``
+        self.a = np.ones(10).astype(config.floatX)
```

Comment:
I'm not sure if it is garanty the tests a run in a fixed order. To be safe, don't override this instance attribute in the test.